Code/Tools/AssetProcessor/native/unittests/AssetScannerUnitTests.cpp

```diff
@@ -7,130 +7,145 @@
-    expectedFiles << tempPath.absoluteFilePath("rootfile1.txt");
-
-    for (const QString& expect : expectedFiles)
+    TEST_F(AssetScannerUnitTest, AssetScanner_ScanMultipleFolders_ExpectedFilesAndFoldersFound)
     {
-        UNIT_TEST_EXPECT_TRUE(CreateDummyFile(expect));
-    }
-
-    // but we're going to not watch subfolder3 recursively, so... remove these files (even though they're on disk)
-    // if this causes a failure it means that its ignoring the "do not recurse" flag:
-    expectedFiles.remove(tempPath.absoluteFilePath("subfolder3/aaa/basefile.txt"));
-    expectedFiles.remove(tempPath.absoluteFilePath("subfolder3/aaa/bbb/basefile.txt"));
-    expectedFiles.remove(tempPath.absoluteFilePath("subfolder3/aaa/bbb/ccc/basefile.txt"));
-
-    QSet<QString> expectedFolders;
-    expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa");
-    expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa/bbb");
-    expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa/bbb/ccc");
-    expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa/bbb/ccc/ddd");
-    expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa/bbb/ccc/ddd/eee.fff.ggg");
-
-
-    PlatformConfiguration config;
-    AZStd::vector<AssetBuilderSDK::PlatformInfo> platforms;
-    config.PopulatePlatformsForScanFolder(platforms);
-    //                                               PATH               DisplayName  PortKey outputfolder  root recurse  platforms
-    config.AddScanFolder(ScanFolderInfo(tempPath.absolutePath(),         "temp",       "ap1",    true,  false, platforms));  // note:  "Recurse" set to false.
-    config.AddScanFolder(ScanFolderInfo(tempPath.filePath("subfolder1"), "",           "ap2",   false,  true,  platforms));
-    config.AddScanFolder(ScanFolderInfo(tempPath.filePath("subfolder2"), "",           "ap3",   false,  true,  platforms));
-    config.AddScanFolder(ScanFolderInfo(tempPath.filePath("subfolder3"), "",           "ap4",   false,  false, platforms)); // note:  "Recurse" set to false.
-    AssetScanner scanner(&config);
-
-    QList<AssetFileInfo> filesFound;
-    QList<AssetFileInfo> foldersFound;
-
-    bool doneScan = false;
-
-    connect(&scanner, &AssetScanner::FilesFound, this, [&filesFound](QSet<AssetFileInfo> fileList)
+        using namespace AssetProcessor;
+        QSharedPointer<QCoreApplication> m_qApp;
+
+        // QApplications require command line arguments, so create something to meet that requirement.
+        int argC{ 1 };
+        char commandLineBuffer[AZ_MAX_PATH_LEN];
+        char* commandLineBufferAddress{ commandLineBuffer };
+        azstrcpy(commandLineBuffer, AZ_ARRAY_SIZE(commandLineBuffer), "no_argv_supplied");
+
+        m_qApp.reset(new QApplication(argC, &commandLineBufferAddress));
+
+        QTemporaryDir tempEngineRoot;
+        QDir tempPath(tempEngineRoot.path());
+
+        QSet<QString> expectedFiles;
```

Comment:
Is that something you're OK with putting off until later? I've got a lot of these tests to convert, so I was hoping to just focus on the conversion to save time.


<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/stankowi/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/stankowi/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>
<!--table
	{mso-displayed-decimal-separator:"\.";
	mso-displayed-thousand-separator:"\,";}
@page
	{margin:.75in .7in .75in .7in;
	mso-header-margin:.3in;
	mso-footer-margin:.3in;}
tr
	{mso-height-source:auto;}
col
	{mso-width-source:auto;}
br
	{mso-data-placement:same-cell;}
td
	{padding-top:1px;
	padding-right:1px;
	padding-left:1px;
	mso-ignore:padding;
	color:black;
	font-size:11.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:Calibri, sans-serif;
	mso-font-charset:0;
	mso-number-format:General;
	text-align:general;
	vertical-align:bottom;
	border:none;
	mso-background-source:auto;
	mso-pattern:auto;
	mso-protection:locked visible;
	white-space:nowrap;
	mso-rotate:0;}
-->
</style>
</head>

<body link="#0563C1" vlink="#954F72">



1,500 lines | AssetProcessingStateDataUnitTests.cpp
-- | --
3,111 lines | AssetProcessorManagerUnitTests.cpp
204 lines | AssetProcessorServerUnitTests.cpp
318 lines | AssetRequestHandlerUnitTests.cpp
136 lines | AssetScannerUnitTests.cpp
370 lines | BuilderSDKUnitTests.cpp
242 lines | ConnectionManagerUnitTests.cpp
43 lines | ConnectionUnitTests.cpp
319 lines | FileWatcherUnitTests.cpp
236 lines | PlatformConfigurationUnitTests.cpp
773 lines | RCcontrollerUnitTests.cpp
554 lines | UtilitiesUnitTests.cpp



</body>

</html>

Reply:
I am fine with saving it for later.

Reply:
We decided that we're going to treat updating all tests as a long term plan, so I went back and addressed this feedback instead of trying to get through all the tests back to back.